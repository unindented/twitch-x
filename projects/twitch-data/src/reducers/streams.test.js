import reducer from './streams'

const topStreamsFixture = require('../__fixtures__/topStreams.json')
const searchStreamsFixture = require('../__fixtures__/searchStreams.json')

describe('reducers/streams', () => {
  describe('default', () => {
    it('returns initial state', () => {
      const prevState = undefined
      expect(reducer(prevState)).toMatchSnapshot()
    })
  })

  describe('LOAD_TOP_STREAMS_SUCCESS', () => {
    it('stores the streams', () => {
      const prevState = {
        byId: {
          dota2ti: {
            name: 'dota2ti'
          }
        }
      }
      const action = {
        type: 'LOAD_TOP_STREAMS_SUCCESS',
        payload: {response: topStreamsFixture}
      }
      expect(reducer(prevState, action)).toMatchSnapshot()
    })
  })

  describe('LOAD_SEARCH_STREAMS_SUCCESS', () => {
    it('stores the streams', () => {
      const prevState = {
        byId: {
          dota2ti: {
            name: 'dota2ti'
          }
        }
      }
      const action = {
        type: 'LOAD_SEARCH_STREAMS_SUCCESS',
        payload: {response: searchStreamsFixture}
      }
      expect(reducer(prevState, action)).toMatchSnapshot()
    })
  })
})
