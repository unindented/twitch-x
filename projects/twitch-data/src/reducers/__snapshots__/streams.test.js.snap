// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/streams LOAD_SEARCH_STREAMS_SUCCESS stores the streams 1`] = `
Object {
  "byId": Object {
    "dota2ti": Object {
      "name": "dota2ti",
    },
    "esl_sc2": Object {
      "channel": "ESL_SC2",
      "game": "StarCraft II",
      "id": "esl_sc2",
      "image": "https://static-cdn.jtvnw.net/previews-ttv/live_user_esl_sc2-{width}x{height}.jpg",
      "status": "RERUN: Welmu vs. Elazer (PvZ) - WCS Season 3 Challenger EU - Match 12",
      "viewers": 327,
    },
    "lowkotv": Object {
      "channel": "LowkoTV",
      "game": "StarCraft II",
      "id": "lowkotv",
      "image": "https://static-cdn.jtvnw.net/previews-ttv/live_user_lowkotv-{width}x{height}.jpg",
      "status": "Lowko :: StarCraft Dreams :: !prime",
      "viewers": 537,
    },
    "ogamingsc2": Object {
      "channel": "OgamingSC2",
      "game": "StarCraft II",
      "id": "ogamingsc2",
      "image": "https://static-cdn.jtvnw.net/previews-ttv/live_user_ogamingsc2-{width}x{height}.jpg",
      "status": "MarathOG - La suite du MarathOG",
      "viewers": 303,
    },
    "wintergaming": Object {
      "channel": "wintergaming",
      "game": "StarCraft II",
      "id": "wintergaming",
      "image": "https://static-cdn.jtvnw.net/previews-ttv/live_user_wintergaming-{width}x{height}.jpg",
      "status": "!Vodcast ❄ New sc2 players !startsc2 + !Guides for !terran/!zerg/!protoss  !youtube :)  Always commentary, never rage*",
      "viewers": 211,
    },
  },
  "search": Array [
    "lowkotv",
    "esl_sc2",
    "ogamingsc2",
    "wintergaming",
  ],
}
`;

exports[`reducers/streams LOAD_TOP_STREAMS_SUCCESS stores the streams 1`] = `
Object {
  "byId": Object {
    "dota2ti": Object {
      "name": "dota2ti",
    },
    "imaqtpie": Object {
      "channel": "imaqtpie",
      "game": "League of Legends",
      "id": "imaqtpie",
      "image": "https://static-cdn.jtvnw.net/previews-ttv/live_user_imaqtpie-{width}x{height}.jpg",
      "status": "wait a minute is this imaqtpies stream the one and only i gotta get here and watch him play league this guy does what no one else can hes so",
      "viewers": 19208,
    },
    "izakooo": Object {
      "channel": "IzakOOO",
      "game": "Counter-Strike: Global Offensive",
      "id": "izakooo",
      "image": "https://static-cdn.jtvnw.net/previews-ttv/live_user_izakooo-{width}x{height}.jpg",
      "status": "[PGL Major] 18:00 BIG vs. Cloud9",
      "viewers": 31147,
    },
    "pgl": Object {
      "channel": "PGL",
      "game": "Counter-Strike: Global Offensive",
      "id": "pgl",
      "image": "https://static-cdn.jtvnw.net/previews-ttv/live_user_pgl-{width}x{height}.jpg",
      "status": "BIG vs Cloud9 (bo1) |  PGL Major Kraków Group Stage",
      "viewers": 277232,
    },
    "starladder5": Object {
      "channel": "Starladder5",
      "game": "Counter-Strike: Global Offensive",
      "id": "starladder5",
      "image": "https://static-cdn.jtvnw.net/previews-ttv/live_user_starladder5-{width}x{height}.jpg",
      "status": "LIVE: Vega Squadron vs Immortals | Round 2 | PGL Major Kraków Group Stage | by Sl4m & Strike",
      "viewers": 55614,
    },
  },
  "top": Array [
    "pgl",
    "starladder5",
    "izakooo",
    "imaqtpie",
  ],
}
`;

exports[`reducers/streams default returns initial state 1`] = `
Object {
  "byId": Object {},
  "search": Array [],
  "top": Array [],
}
`;
